### PR TITLE
Pull in nodenv-package-rehash plugin

### PR DESCRIPTION
**Pull request by jasonkarns**
As mentioned in https://github.com/nodenv/nodenv-package-rehash/issues/8, the
rbenv-gem-rehash plugin (of which this plugin is a spiritual descendant) 
was incorporated into rbenv-core. This is the beginning stages of a similar move.

Issues:
 - yarn does not seem to invoke postinstall/postuninstall hooks when packages are installed globally
 - need to pull over tests
 - npm is _super_ noisy when executing these hooks because it invokes the hook for each _dependency_ of the globally installed/uninstalled package. Would like to resolve this before merging.
 - need to figure out how to symlink from the nodenv version into homebrew's Cellar in such a way that the symlink is kept up to date when nodenv is upgraded.

### CODE REVIEW

**Comment by jasonkarns** [comment]
Incidentally, this move should help un-block https://github.com/nodenv/node-build/issues/237; which is more complicated than it should be because of the need for plugins to know the PREFIX_PATH (or attempt to derive it from $VERSION_NAME)

**Comment by AlJohri** [comment]
hey @jasonkarns, I haven't looked into this too deeply, but I was wondering if you looked into how pyenv does its auto rehash?

https://github.com/pyenv/pyenv/blob/master/pyenv.d/exec/pip-rehash/pip

I noticed you wrote

> npm is super noisy when executing these hooks because it invokes the hook for each dependency of the globally installed/uninstalled package

Perhaps the `Remove pyenv-pip-rehash/libexec from PATH to avoid infinite loops` part of the script above could help?

**Comment by AlJohri** [comment]
hey @jasonkarns, any update here?

**Comment by neersighted** [comment]
Anything that can be done to get this merged? I'd be willing to take over this PR if @jasonkarns is otherwise occupied.

**Comment by jasonkarns** [comment]
Closing for the reasons stated here: https://github.com/nodenv/nodenv-package-rehash/issues/8#issuecomment-456647838

TL;DR:

npm <5 is _WAY_ too noisy when invoking global lifecycle hooks;
npm <5 is bundled in all nodes <8, which still impacts a too-large group of users.
That's too much of a negative experience when installing any global module for this to be baked into core (for now). I think the cost/benefit ratio will shift nearer the end of 2019 when many fewer people will be on nodes < 8.
